include/sound/soc-dai.h

```diff
@@ -468,6 +468,8 @@ struct snd_soc_dai {
 
 	/* bit field */
 	unsigned int probed:1;
+
+	void *private; /* core does not touch this */
```

Comment:
I am not a big fan of using drvdata @ujfalusi , that's what we already use for the top-level information. 

````
	/* set driver data, accessed by snd_soc_dai_get_drvdata() */
	auxiliary_set_drvdata(auxdev, cdns);
````

If we have to nest structures in there, it's not that elegant.

I also don't get why we would need to special-case playback or capture. dai_drvdata is actually a good counter-example of information that can be used by the DAI but is not direction specific.

The case of dma_data with playback or capture is IMHO pushing it, it's the exception to the rule rather than the rule to follow. If your DAI can support two directions, you can take care of the direction within the private data. There's no need to allocate two pointers.

I wasn't able to understand the comment "would rather have separate dai_data from the start to make sure that others can use it".  If you are suggesting two pointers for playback and capture, it would make things more confusing IMHO. When you give people two alternatives, e.g. between playback_dma_data and playback_dai_data, then half of the world will get things 'wrong'.